dali/operators/generic/slice/slice_attr.h

```diff
@@ -33,28 +33,37 @@
 
 namespace dali {
 
-class SliceAttr {
+class NamedSliceAttr {
  public:
-  explicit inline SliceAttr(const OpSpec &spec)
+  explicit inline NamedSliceAttr(const OpSpec &spec,
+                                 const char* start_name = "start",
+                                 const char* rel_start_name = "rel_start",
+                                 const char* end_name = "end",
+                                 const char* rel_end_name = "rel_end",
+                                 const char* shape_name = "shape",
+                                 const char* rel_shape_name = "rel_shape",
+                                 const char* axes_name = "axes",
+                                 const char* axis_names_name = "axis_names")
       : spec_(spec),
-        normalized_anchor_(spec.GetArgument<bool>("normalized_anchor")),
-        normalized_shape_(spec.GetArgument<bool>("normalized_shape")),
-        start_("start", spec),
-        rel_start_("rel_start", spec),
-        end_("end", spec),
-        rel_end_("rel_end", spec),
-        shape_("shape", spec),
-        rel_shape_("rel_shape", spec),
-        crop_window_generators_(spec.GetArgument<int>("max_batch_size")) {
-    const bool has_axes_arg = spec.HasArgument("axes");
-    const bool has_axis_names_arg = spec.HasArgument("axis_names");
+        start_(start_name, spec),
+        rel_start_(rel_start_name, spec),
+        end_(end_name, spec),
+        rel_end_(rel_end_name, spec),
+        shape_(shape_name, spec),
+        rel_shape_(rel_shape_name, spec) {
+    int max_batch_sz = spec.GetArgument<int>("max_batch_size");
+    crop_window_generators_.resize(max_batch_sz);
+    const bool use_axes = axes_name != nullptr;
+    const bool use_axis_names = axis_names_name != nullptr;
+    const bool has_axes_arg = use_axes && spec.HasArgument(axes_name);
+    const bool has_axis_names_arg = use_axis_names && spec.HasArgument(axis_names_name);
```

Comment:
What if both `axes_name` and `axis_names_name` are provided?

Reply:
In this case axis_names has higher priority, as it always had.
```
If a value is provided, ``axis_names`` will have a higher priority than ``axes``.
    ````

Reply:
ok